File: src/main/java/com/youngjun/auth/core/api/controller/v1/TokenController.java
package com.youngjun.auth.core.api.controller.v1;

import com.youngjun.auth.core.api.controller.v1.request.ReissueTokenRequestDto;
import com.youngjun.auth.core.api.controller.v1.response.ReissueTokenResponseDto;
import com.youngjun.auth.core.api.support.response.AuthResponse;
import com.youngjun.auth.core.domain.token.TokenPair;
import com.youngjun.auth.core.domain.token.TokenService;
import jakarta.validation.Valid;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RestController;

@RestController
public class TokenController {

    private final TokenService tokenService;

    public TokenController(TokenService tokenService) {
        this.tokenService = tokenService;
    }

    @PostMapping("/auth/token")
    public AuthResponse<ReissueTokenResponseDto> reissue(@RequestBody @Valid ReissueTokenRequestDto request) {
        TokenPair tokenPair = tokenService.reissue(request.refreshToken());
        return AuthResponse.success(ReissueTokenResponseDto.from(tokenPair));
    }

}

File: src/main/kotlin/com/youngjun/auth/core/api/controller/v1/TokenController.kt
package com.youngjun.auth.core.api.controller.v1

import com.youngjun.auth.core.api.application.TokenService
import com.youngjun.auth.core.api.controller.v1.request.ReissueTokenRequest
import com.youngjun.auth.core.api.controller.v1.response.ReissueTokenResponse
import com.youngjun.auth.core.api.support.response.AuthResponse
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RestController

@RestController
class TokenController(
    private val tokenService: TokenService,
) {
    @PostMapping("/auth/token")
    fun reissue(
        @RequestBody request: ReissueTokenRequest,
    ): AuthResponse<ReissueTokenResponse> {
        val tokenPair = tokenService.reissue(request.toRefreshToken())
        return AuthResponse.success(ReissueTokenResponse.from(tokenPair))
    }
}
